In [14]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [15]:
import pandas as pd
import numpy as np
import polars as pl

import matplotlib.pyplot as plt
import seaborn as sns

from src.data import make_dataset
from src.features import build_features
from src.visualization import visualize

sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_A_ANLIZAR = 112103  # Este es el Torax

## Obtencion de Métricas de Egresos Hospitalarios

En este apartado se quieren calculcar diversas métricas relevantes para el análisis de los
egresos hospitalarios en Chile. Las métricas a calcular son las siguientes:

1. Cantidad de Egresos Hospitalarios
2. Dias de Estada Totales
3. Cantidad de Intervenciones Quirúrgicas
4. Cantidad de Muertos

Todas estas métricas serán calculadas para cada Año, Establecimiento de Salud y Diagnóstico
distinto que exista en la base de datos. Por ejemplo:

"En el año 2019, el Hospital San José (código XXXXXX) tuvo XX egresos, con XX dias estada promedio,
con XX intervenciones quirúrgicas y XX muertos para el diagnóstico A24.4"s


In [16]:
df_nacional = pl.scan_csv(
    "../data/processed/egresos_procesados.csv",
    infer_schema_length=10000,
    null_values="Extranjero",
)

In [17]:
AGRUPACION = [
    "ANO_EGRESO",
    "ESTABLECIMIENTO_SALUD",
    "GLOSA_ESTABLECIMIENTO_SALUD",
    "DIAG1",
]

metricas = build_features.obtener_metricas_egresos(df_nacional, AGRUPACION).collect()

In [18]:
display(metricas.sample(10))

shape: (10, 8)
┌────────────┬──────────────┬──────────────┬───────┬───────────┬─────────────┬─────────┬───────────┐
│ ANO_EGRESO ┆ ESTABLECIMIE ┆ GLOSA_ESTABL ┆ DIAG1 ┆ n_egresos ┆ dias_estada ┆ n_int_q ┆ n_muertos │
│ ---        ┆ NTO_SALUD    ┆ ECIMIENTO_SA ┆ ---   ┆ ---       ┆ _totales    ┆ ---     ┆ ---       │
│ i64        ┆ ---          ┆ LUD          ┆ str   ┆ u32       ┆ ---         ┆ i64     ┆ i64       │
│            ┆ i64          ┆ ---          ┆       ┆           ┆ i64         ┆         ┆           │
│            ┆              ┆ str          ┆       ┆           ┆             ┆         ┆           │
╞════════════╪══════════════╪══════════════╪═══════╪═══════════╪═════════════╪═════════╪═══════════╡
│ 2007       ┆ 118202       ┆ Clínica de   ┆ P923  ┆ 6         ┆ 13          ┆ 0       ┆ 0         │
│            ┆              ┆ la Mujer     ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Sanatorio    ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Al…          ┆       ┆           ┆             ┆         ┆           │
│ 2011       ┆ 107100       ┆ Hospital Dr. ┆ L538  ┆ 1         ┆ 5           ┆ 0       ┆ 0         │
│            ┆              ┆ Gustavo      ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Fricke (Viñ… ┆       ┆           ┆             ┆         ┆           │
│ 2020       ┆ 121119       ┆ Hospital de  ┆ H812  ┆ 1         ┆ 10          ┆ 0       ┆ 0         │
│            ┆              ┆ Gorbea       ┆       ┆           ┆             ┆         ┆           │
│ 2018       ┆ 114103       ┆ Hospital     ┆ S625  ┆ 1         ┆ 1           ┆ 1       ┆ 0         │
│            ┆              ┆ Padre        ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Alberto      ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Hurtado (…   ┆       ┆           ┆             ┆         ┆           │
│ …          ┆ …            ┆ …            ┆ …     ┆ …         ┆ …           ┆ …       ┆ …         │
│ 2015       ┆ 110150       ┆ Hospital San ┆ J159  ┆ 6         ┆ 33          ┆ 0       ┆ 0         │
│            ┆              ┆ José         ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ (Melipilla)  ┆       ┆           ┆             ┆         ┆           │
│ 2004       ┆ 102100       ┆ Hospital Dr. ┆ T07X  ┆ 5         ┆ 8           ┆ 4       ┆ 0         │
│            ┆              ┆ Ernesto      ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Torres Gald… ┆       ┆           ┆             ┆         ┆           │
│ 2016       ┆ 112101       ┆ Hospital Dr. ┆ O318  ┆ 1         ┆ 3           ┆ 0       ┆ 0         │
│            ┆              ┆ Luis Tisné   ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ B. (Sant…    ┆       ┆           ┆             ┆         ┆           │
│ 2012       ┆ 112201       ┆ Clínica      ┆ E848  ┆ 2         ┆ 2           ┆ 0       ┆ 0         │
│            ┆              ┆ Tabancura    ┆       ┆           ┆             ┆         ┆           │
└────────────┴──────────────┴──────────────┴───────┴───────────┴─────────────┴─────────┴───────────┘

- Al ver una muestra del calculo realizado, se puede observar la cantidad de egresos, dias de
  estada, numero de intervenciones quirurgicas y cantidad de muertos para 10 duplas de
  hospitales - diagnosticos distintos.


Si bien se tienen las metricas calculadas, ahora se quiere responder a las siguientes preguntas:

- En qué diagnósticos se destaca cada hospital productivamente? Esto significa, en qué diagnósticos
  cada hospital tiene una mayor cantidad de egresos que otros hospitales?

- El hospital tiene un desempeño similar dentro de distintos grupos/estratos de hospitales?
  El hospital se desempeñara de la misma forma nacionalmente, dentro del estrato público, privado,
  etc...?

Para responder a la pregunta se realizará un ranking de todos los hospitales para cada
diagnóstico y cada año observado. Este ranking se realizará para distintos grupos/estratos de
hospitales (Nacional, Hospitales Públicos, Hospitales Privados, Hospitales GRD)


## Análisis de Estratos


In [19]:
dict_estratos = build_features.obtener_diccionario_estratos(df_nacional, HOSPITAL_A_ANLIZAR)

In [20]:
variables_a_rankear = ["n_egresos"]
subgrupo_del_ranking = ["ANO_EGRESO", "DIAG1"]

rankings_nacionales = build_features.agregar_ranking_estratos(
    metricas, dict_estratos, variables_a_rankear, subgrupo_del_ranking
)

rankings_hospital_analizado = rankings_nacionales.filter(
    pl.col("ESTABLECIMIENTO_SALUD") == HOSPITAL_A_ANLIZAR
)

In [21]:
diags_mas_relevantes = rankings_hospital_analizado.filter(
    (pl.col("ranking_nacionales_n_egresos") == 1)
    & (pl.col("n_int_q") > 0)
    & (pl.col("ANO_EGRESO") > 2011)
).sort(by=["ANO_EGRESO", "n_egresos", "n_int_q"], descending=True)

veinte_mas_relevantes_por_anio = diags_mas_relevantes.groupby(["ANO_EGRESO"]).head(20)

C:\Users\ppizarro\AppData\Local\Temp\ipykernel_16792\516069765.py:7: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.
  veinte_mas_relevantes_por_anio = diags_mas_relevantes.groupby(["ANO_EGRESO"]).head(20)


In [22]:
DIAGS_MAS_RELEVANTES = veinte_mas_relevantes_por_anio.select(pl.col("DIAG1").unique()).sort(
    by="DIAG1"
)

In [23]:
print(DIAGS_MAS_RELEVANTES)

shape: (45, 1)
┌───────┐
│ DIAG1 │
│ ---   │
│ str   │
╞═══════╡
│ C33X  │
│ C340  │
│ C341  │
│ C342  │
│ …     │
│ Q231  │
│ Q676  │
│ T820  │
│ Z450  │
└───────┘


In [24]:
DIAGS_1_o_2_en_2019 = (
    rankings_hospital_analizado.filter(
        (pl.col("DIAG1").is_in(DIAGS_MAS_RELEVANTES.to_series()))
        & (pl.col("ANO_EGRESO") == 2019)
        & (
            (pl.col("ranking_nacionales_n_egresos") == 1)
            | (pl.col("ranking_nacionales_n_egresos") == 2)
        )
    )
    .select(pl.col("DIAG1"))
)

In [25]:
print(DIAGS_1_o_2_en_2019)

shape: (33, 1)
┌───────┐
│ DIAG1 │
│ ---   │
│ str   │
╞═══════╡
│ Z450  │
│ T820  │
│ M348  │
│ J955  │
│ …     │
│ C342  │
│ C341  │
│ C340  │
│ C33X  │
└───────┘


In [35]:
rankings_hospital_analizado

shape: (5_922, 27)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ ANO_EGRES ┆ ESTABLECI ┆ GLOSA_EST ┆ Capítulo  ┆ … ┆ %_grd_n_e ┆ ranking_i ┆ total_int ┆ %_intern │
│ O         ┆ MIENTO_SA ┆ ABLECIMIE ┆ ---       ┆   ┆ gresos    ┆ nterno_n_ ┆ erno_n_eg ┆ o_n_egre │
│ ---       ┆ LUD       ┆ NTO_SALUD ┆ str       ┆   ┆ ---       ┆ egresos   ┆ resos     ┆ sos      │
│ i64       ┆ ---       ┆ ---       ┆           ┆   ┆ f64       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ i64       ┆ str       ┆           ┆   ┆           ┆ u32       ┆ u32       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ 2020      ┆ 112103    ┆ Instituto ┆ Cap.21    ┆ … ┆ 0.002755  ┆ 253       ┆ 2848      ┆ 0.000351 │
│           ┆           ┆ Nacional  ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 112103    ┆ Instituto ┆ Cap.21    ┆ … ┆ 0.00761   ┆ 97        ┆ 2848      ┆ 0.001756 │
│           ┆           ┆ Nacional  ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 112103    ┆ Instituto ┆ Cap.21    ┆ … ┆ 0.111111  ┆ 315       ┆ 2848      ┆ 0.000351 │
│           ┆           ┆ Nacional  ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 112103    ┆ Instituto ┆ Cap.21    ┆ … ┆ 0.02459   ┆ 46        ┆ 2848      ┆ 0.005267 │
│           ┆           ┆ Nacional  ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ 2001      ┆ 112103    ┆ Instituto ┆ Cap.01    ┆ … ┆ 0.048077  ┆ 73        ┆ 4985      ┆ 0.001003 │
│           ┆           ┆ Nacional  ┆ CIERTAS   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ DES INF…  ┆   ┆           ┆           ┆           ┆          │
│ 2001      ┆ 112103    ┆ Instituto ┆ Cap.01    ┆ … ┆ 0.142237  ┆ 10        ┆ 4985      ┆ 0.026279 │
│           ┆           ┆ Nacional  ┆ CIERTAS   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ DES INF…  ┆   ┆           ┆           ┆           ┆          │
│ 2001      ┆ 112103    ┆ Instituto ┆ Cap.01    ┆ … ┆ 0.055556  ┆ 176       ┆ 4985      ┆ 0.000201 │
│           ┆           ┆ Nacional  ┆ CIERTAS   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Enferm ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ edad…     ┆ DES 

In [71]:
tabla_resumen_egresos_torax_relevantes = (
    rankings_hospital_analizado.filter(
        ((pl.col("ANO_EGRESO") >= 2016) & (pl.col("ANO_EGRESO") <= 2019))
        & (pl.col("DIAG1").is_in(DIAGS_1_o_2_en_2019.to_series()))
    )
    .to_pandas()
    .pivot(
        index=["DIAG1", "Descripción"],
        columns="ANO_EGRESO",
        values=[
            "ranking_nacionales_n_egresos",
            "n_egresos",
            "total_nacionales_n_egresos",
            "%_nacionales_n_egresos",
        ],
    )
).sort_values(
    by=[("ranking_nacionales_n_egresos", 2019), ("n_egresos", 2019)], ascending=[True, False]
)

In [65]:
resumen_egresos_torax_y_nacional_relevantes = (
    tabla_resumen_egresos_torax_relevantes[("n_egresos")].astype(int).astype(str)
    + " ("
    + tabla_resumen_egresos_torax_relevantes[("total_nacionales_n_egresos")].astype(int).astype(str)
    + "; "
    + round((tabla_resumen_egresos_torax_relevantes[("%_nacionales_n_egresos")] * 100), 1).astype(
        str
    )
    + ")"
)

posiciones_torax_egresos_relevantes = tabla_resumen_egresos_torax_relevantes[
    ("ranking_nacionales_n_egresos")
].astype(int)

resumen_egresos_y_posiciones_relevantes = pd.concat(
    [resumen_egresos_torax_y_nacional_relevantes, posiciones_torax_egresos_relevantes], axis=1
)

display(resumen_egresos_y_posiciones_relevantes)

ANO_EGRESO                                                           2016  \
DIAG1 Descripción                                                           
C33X  Tumor maligno de la tráquea                            4 (20; 20.0)   
C340  Tumor maligno del bronquio principal                 21 (144; 14.6)   
C341  Tumor maligno del lóbulo superior, bronquio o p...  180 (400; 45.0)   
C342  Tumor maligno del lóbulo medio, bronquio o pulmón     25 (53; 47.2)   
C343  Tumor maligno del lóbulo inferior, bronquio o p...  120 (241; 49.8)   
C381  Tumor maligno del mediastino anterior                  9 (26; 34.6)   
C384  Tumor maligno de la pleura                            17 (55; 30.9)   
C450  Mesotelioma de la pleura                              22 (65; 33.8)   
C780  Tumor maligno secundario del pulmón                  43 (328; 13.1)   
C782  Tumor maligno secundario de la pleura                42 (136; 30.9)   
D143  Tumor benigno de los bronquios y del pulmón          21 (189; 11.1)   
D381  Tumor de comportamiento incierto o desconocido ...   91 (637; 14.3)   
E848  Fibrosis quística con otras manifestaciones           23 (51; 45.1)   
I080  Trastornos de las válvulas mitral y aórtica          35 (112; 31.2)   
I081  Trastornos de las válvulas mitral y tricúspide        43 (64; 67.2)   
I340  Insuficiencia (de la válvula) mitral                  39 (392; 9.9)   
I351  Insuficiencia (de la válvula) aórtica                 19 (209; 9.1)   
I352  Estenosis (de la válvula) aórtica con insuficie...    35 (94; 37.2)   
I456  Síndrome de preexcitación                            31 (190; 16.3)   
I472  Taquicardia ventricular                               14 (356; 3.9)   
I495  Síndrome del seno enfermo                             68 (756; 9.0)   
I712  Aneurisma de la aorta torácica, sin mención de ...    13 (94; 13.8)   
J47X  Bronquiectasia                                       58 (370; 15.7)   
J679  Neumonitis debida a hipersensibilidad a polvo o...    53 (68; 77.9)   
J841  Otras enfermedades pulmonares intersticiales co...  124 (1617; 7.7)   
J848  Otras enfermedades pulmonares intersticiales es...    40 (98; 40.8)   
J849  Enfermedad pulmonar intersticial, no especificada     36 (556; 6.5)   
J860  Piotórax con fístula                                  15 (49; 30.6)   
J869  Piotórax sin fístula                                  19 (334; 5.7)   
J955  Estenosis subglótica consecutiva a procedimientos    36 (124; 29.0)   
M348  Otras formas de esclerosis sistémica                  17 (37; 45.9)   
T820  Complicación mecánica de prótesis de válvula ca...     8 (39; 20.5)   
Z450  Asistencia y ajuste de marcapaso cardíaco           101 (1146; 8.8)   

ANO_EGRESO                                                           2017  \
DIAG1 Descripción                                                           
C33X  Tumor maligno de la tráquea                             1 (25; 4.0)   
C340  Tumor maligno del bronquio principal                 21 (163; 12.9)   
C341  Tumor maligno del lóbulo superior, bronquio o p...  180 (372; 48.4)   
C342  Tumor maligno del lóbulo medio, bronquio o pulmón     19 (50; 38.0)   
C343  Tumor maligno del lóbulo inferior, bronquio o p...  118 (258; 45.7)   
C381  Tumor maligno del mediastino anterior                 15 (41; 36.6)   
C384  Tumor maligno de la pleura                             9 (39; 23.1)   
C450  Mesotelioma de la pleura                              23 (81; 28.4)   
C780  Tumor maligno secundario del pulmón                  57 (373; 15.3)   
C782  Tumor maligno secundario de la pleura                29 (147; 19.7)   
D143  Tumor benigno de los bronquios y del pulmón          32 (178; 18.0)   
D381  Tumor de comportamiento incierto o desconocido ...    36 (553; 6.5)   
E848  Fibrosis quística con otras manifestaciones           26 (58; 44.8)   
I080  Trastornos de las válvulas mitral y aórtica           38 (96; 39.6)   
I081  Trastornos de las válvulas mitral y tricúspide        29 (51; 56.9

ANO_EGRESO                                                           2016  \
DIAG1 Descripción                                                           
C33X  Tumor maligno de la tráquea                            4 (20; 20.0)   
C340  Tumor maligno del bronquio principal                 21 (144; 14.6)   
C341  Tumor maligno del lóbulo superior, bronquio o p...  180 (400; 45.0)   
C342  Tumor maligno del lóbulo medio, bronquio o pulmón     25 (53; 47.2)   
C343  Tumor maligno del lóbulo inferior, bronquio o p...  120 (241; 49.8)   
C381  Tumor maligno del mediastino anterior                  9 (26; 34.6)   
C384  Tumor maligno de la pleura                            17 (55; 30.9)   
C450  Mesotelioma de la pleura                              22 (65; 33.8)   
C780  Tumor maligno secundario del pulmón                  43 (328; 13.1)   
C782  Tumor maligno secundario de la pleura                42 (136; 30.9)   
D143  Tumor benigno de los bronquios y del pulmón          21 (189; 11.1)   
D381  Tumor de comportamiento incierto o desconocido ...   91 (637; 14.3)   
E848  Fibrosis quística con otras manifestaciones           23 (51; 45.1)   
I080  Trastornos de las válvulas mitral y aórtica          35 (112; 31.2)   
I081  Trastornos de las válvulas mitral y tricúspide        43 (64; 67.2)   
I340  Insuficiencia (de la válvula) mitral                  39 (392; 9.9)   
I351  Insuficiencia (de la válvula) aórtica                 19 (209; 9.1)   
I352  Estenosis (de la válvula) aórtica con insuficie...    35 (94; 37.2)   
I456  Síndrome de preexcitación                            31 (190; 16.3)   
I472  Taquicardia ventricular                               14 (356; 3.9)   
I495  Síndrome del seno enfermo                             68 (756; 9.0)   
I712  Aneurisma de la aorta torácica, sin mención de ...    13 (94; 13.8)   
J47X  Bronquiectasia                                       58 (370; 15.7)   
J679  Neumonitis debida a hipersensibilidad a polvo o...    53 (68; 77.9)   
J841  Otras enfermedades pulmonares intersticiales co...  124 (1617; 7.7)   
J848  Otras enfermedades pulmonares intersticiales es...    40 (98; 40.8)   
J849  Enfermedad pulmonar intersticial, no especificada     36 (556; 6.5)   
J860  Piotórax con fístula                                  15 (49; 30.6)   
J869  Piotórax sin fístula                                  19 (334; 5.7)   
J955  Estenosis subglótica consecutiva a procedimientos    36 (124; 29.0)   
M348  Otras formas de esclerosis sistémica                  17 (37; 45.9)   
T820  Complicación mecánica de prótesis de válvula ca...     8 (39; 20.5)   
Z450  Asistencia y ajuste de marcapaso cardíaco           101 (1146; 8.8)   

ANO_EGRESO                                                           2017  \
DIAG1 Descripción                                                           
C33X  Tumor maligno de la tráquea                             1 (25; 4.0)   
C340  Tumor maligno del bronquio principal                 21 (163; 12.9)   
C341  Tumor maligno del lóbulo superior, bronquio o p...  180 (372; 48.4)   
C342  Tumor maligno del lóbulo medio, bronquio o pulmón     19 (50; 38.0)   
C343  Tumor maligno del lóbulo inferior, bronquio o p...  118 (258; 45.7)   
C381  Tumor maligno del mediastino anterior                 15 (41; 36.6)   
C384  Tumor maligno de la pleura                             9 (39; 23.1)   
C450  Mesotelioma de la pleura                              23 (81; 28.4)   
C780  Tumor maligno secundario del pulmón                  57 (373; 15.3)   
C782  Tumor maligno secundario de la pleura                29 (147; 19.7)   
D143  Tumor benigno de los bronquios y del pulmón          32 (178; 18.0)   
D381  Tumor de comportamiento incierto o desconocido ...    36 (553; 6.5)   
E848  Fibrosis quística con otras manifestaciones           26 (58; 44.8)   
I080  Trastornos de las válvulas mitral y aórtica           38 (96; 39.6)   
I081  Trastornos de las válvulas mitral y tricúspide        29 (51; 56.9

In [ ]:
visualize.add_dataframes_to_powerpoint(
    [resumen_torax_y_nacional_con_egresos],
    "../reports/presentacion.pptx",
    font_size=9,
    cell_width=12,
    cell_height=0.5,
    font_family="Open Sans",
    max_cell_characters=100,
)

Para hacer un contorl de lo que se calculo, se observará una muestra de 10 hospitales - diagnósticos
distintos


In [ ]:
display(rankings_nacionales.sample(10))

shape: (10, 27)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ ANO_EGRES ┆ ESTABLECI ┆ GLOSA_EST ┆ Capítulo  ┆ … ┆ %_grd_n_e ┆ ranking_i ┆ total_int ┆ %_intern │
│ O         ┆ MIENTO_SA ┆ ABLECIMIE ┆ ---       ┆   ┆ gresos    ┆ nterno_n_ ┆ erno_n_eg ┆ o_n_egre │
│ ---       ┆ LUD       ┆ NTO_SALUD ┆ str       ┆   ┆ ---       ┆ egresos   ┆ resos     ┆ sos      │
│ i64       ┆ ---       ┆ ---       ┆           ┆   ┆ f64       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ i64       ┆ str       ┆           ┆   ┆           ┆ u32       ┆ u32       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ 2008      ┆ 117101    ┆ Hospital  ┆ Cap.18    ┆ … ┆ 0.1       ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Clínico   ┆ SÍNTOMAS, ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Herminda  ┆ SIGNOS Y  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Martín…   ┆ RESUL…    ┆   ┆           ┆           ┆           ┆          │
│ 2017      ┆ 112528    ┆ Clínica   ┆ Cap.17    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Las Lilas ┆ MALFORMAC ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ IONES CON ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ GÉNITA…   ┆   ┆           ┆           ┆           ┆          │
│ 2005      ┆ 121119    ┆ Hospital  ┆ Cap.04    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de Gorbea ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ DES ENDOC ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ RINAS,…   ┆   ┆           ┆           ┆           ┆          │
│ 2017      ┆ 114105    ┆ Hospital  ┆ Cap.10    ┆ … ┆ 0.029412  ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Clínico   ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Metropoli ┆ DES DEL   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ tano L…   ┆ APARATO…  ┆   ┆           ┆           ┆           ┆          │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ 2018      ┆ 121202    ┆ Clínica   ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Alemana   ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Temuco ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2019      ┆ 110270    ┆ Clínica   ┆ Cap.09    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Los       ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Maitenes  ┆ DES DEL   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ APARATO…  ┆   ┆           ┆           ┆           ┆          │
│ 2019      ┆ 109201    ┆ Clínica   ┆ Cap.13    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Dávila    ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ DES DEL   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ APARATO…  ┆   ┆           ┆           ┆           ┆          │
│ 2014      ┆ 112528    ┆ Clínica   ┆ Cap.02    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Las Lilas ┆ NEOPLASIA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ S      

En la muestra anterior se puede observar las métricas previamente calculada, y el ranking
respectivo que tiene cada hospital para diagnósticos en cada año específico.


In [ ]:
rankings_nacionales.to_pandas().to_csv(
    "../data/interim/ranking_nacional_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

rankings_hospital_analizado.to_pandas().to_csv(
    "../data/interim/ranking_torax_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

## Metricas para personas sobre 14 años

En este análisis se quiere ver el ranking de egresos pero solamente tomando en cuenta a personas
mayores a 14 años. Esto, para dilucidar si el hospital del tórax es altamente relevante para tratar
patologías congénitas de adultos.


In [ ]:
df_nacional_mayores_a_14 = df_nacional.filter(pl.col("EDAD_A_OS") > 14)

In [ ]:
metricas_sobre_14 = build_features.obtener_metricas_egresos(
    df_nacional_mayores_a_14, AGRUPACION
).collect()

In [ ]:
variables_a_rankear = ["n_egresos"]
subgrupo_del_ranking = ["ANO_EGRESO", "DIAG1"]

rankings_nacionales_sobre_14 = build_features.agregar_ranking_estratos(
    metricas_sobre_14, dict_estratos, variables_a_rankear, subgrupo_del_ranking
)

diags_congenitos_relevantes = rankings_nacionales_sobre_14.filter(
    (pl.col("DIAG1").is_in(["Q230", "Q211", "Q231"])) & (pl.col("ANO_EGRESO") == 2019)
)

In [ ]:
diags_congenitos_relevantes.to_pandas().to_csv(
    "../data/interim/diags_congenitas_sobre_14_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

## 3. Análisis de procedimientos DEIS

En este análisis se quieren responder las siguientes preguntas:

1. ¿Cuál es el procedimiento más frecuente dentro de los pacientes que se les realizó un procedimiento
   quirúrgico?


In [ ]:
proced = (
    df_nacional.filter(pl.col("INTERV_Q") == 1)
    .groupby(["ANO_EGRESO", "ESTABLECIMIENTO_SALUD", "GLOSA_PROCED_PPAL"])
    .agg(pl.count())
).collect()

C:\Users\ppizarro\AppData\Local\Temp\ipykernel_19856\3397463805.py:3: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.
  .groupby(["ANO_EGRESO", "ESTABLECIMIENTO_SALUD", "GLOSA_PROCED_PPAL"])


In [ ]:
display(proced)

shape: (10_757, 4)
┌────────────┬───────────────────────┬───────────────────────────────────┬───────┐
│ ANO_EGRESO ┆ ESTABLECIMIENTO_SALUD ┆ GLOSA_PROCED_PPAL                 ┆ count │
│ ---        ┆ ---                   ┆ ---                               ┆ ---   │
│ i64        ┆ i64                   ┆ str                               ┆ u32   │
╞════════════╪═══════════════════════╪═══════════════════════════════════╪═══════╡
│ 2001       ┆ 103203                ┆ null                              ┆ 5921  │
│ 2001       ┆ 107212                ┆ null                              ┆ 1504  │
│ 2001       ┆ 106100                ┆ null                              ┆ 9333  │
│ 2001       ┆ 107100                ┆ null                              ┆ 9959  │
│ …          ┆ …                     ┆ …                                 ┆ …     │
│ 2020       ┆ 126201                ┆ Registro ecoarterial o ecovenoso… ┆ 3     │
│ 2020       ┆ 118202                ┆ Tomografía Computarizada de Colu… ┆ 1     │
│ 2020       ┆ 200486                ┆ Cistoscopia con sondeo de uno o … ┆ 1     │
│ 2020       ┆ 105101                ┆ Histeroscopía diagnostica o tera… ┆ 1     │
└────────────┴───────────────────────┴───────────────────────────────────┴───────┘

Los resultados muestran el desglose por año y recinto hospitalario junto a la cantidad de
procedimientos realizados en cada intervención quirúrgica. Sin embargo, se puede observar que
existe una cantidad importante de casos donde la glosa del procedimiento fue nula.

Ahora, se quiere calcular cuántas intervenciones quirúrgicas carecen de una glosa de procedimiento.


In [ ]:
display(proced.groupby("GLOSA_PROCED_PPAL").agg(pl.col("count").sum()).sort("count"))

C:\Users\ppizarro\AppData\Local\Temp\ipykernel_19856\989951252.py:1: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.
  display(proced.groupby("GLOSA_PROCED_PPAL").agg(pl.col("count").sum()).sort("count"))


shape: (422, 2)
┌───────────────────────────────────┬──────────┐
│ GLOSA_PROCED_PPAL                 ┆ count    │
│ ---                               ┆ ---      │
│ str                               ┆ u32      │
╞═══════════════════════════════════╪══════════╡
│ Enema baritado del colon (incluy… ┆ 1        │
│ tonometria ocular, cualquier tec… ┆ 1        │
│ Histerosalpingografía (a.c. 04-0… ┆ 1        │
│ Cócleovestibular con electronist… ┆ 1        │
│ …                                 ┆ …        │
│ Instilacion vesical (incluye col… ┆ 7685     │
│ Código no existe actualmente      ┆ 11155    │
│ Apoyo fluoroscópico a procedimie… ┆ 15775    │
│ null                              ┆ 12382642 │
└───────────────────────────────────┴──────────┘

Por lo tanto, 12382642 de todas las intervenciones quirúrgicas carecen de una glosa de procedimiento.


In [ ]:
proced.write_excel("../data/interim/conteo_procedimientos.xlsx")